src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,272 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * <h>duke.Ui</h>
+ * Deals with interactions with the user.
+ */
+public class Ui {
```

Comment:
Very comprehensive set of methods to show error messages and so on. Probably recieving user input here would fit in well with the OOP principles and increase modularity of your code.